app/scripts/modules/core/src/notification/modal/NotificationDetails.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from 'react';
+import { FormikProps } from 'formik';
+import { Option } from 'react-select';
+
+import { INotification, INotificationTypeConfig } from 'core/domain';
+import { Registry } from 'core/registry';
+import { ChecklistInput, FormikFormField, TextAreaInput } from 'core/presentation';
+import { NotificationSelector } from 'core/notification';
+import { NotificationTransformer } from '../notification.transformer';
+
+import './editNotification.less';
+
+export interface INotificationDetailsProps {
+  formik: FormikProps<INotification>;
+  level?: string;
+  stageType?: string;
+}
+
+export interface INotificationDetailsState {
+  notificationTypes: Option[];
+  whenOptions: string[];
+}
+
+export class NotificationDetails extends React.Component<INotificationDetailsProps, INotificationDetailsState> {
+  constructor(props: INotificationDetailsProps) {
+    super(props);
+
+    let whenOptions = [];
+    if (props.level === 'application' || props.level === 'pipeline') {
+      whenOptions = ['pipeline.starting', 'pipeline.complete', 'pipeline.failed'];
+    } else if (props.stageType === 'manualJudgment') {
+      whenOptions = ['manualJudgment', 'manualJudgmentContinue', 'manualJudgmentStop'];
+    } else {
+      whenOptions = ['stage.starting', 'stage.complete', 'stage.failed'];
```

Comment:
these lists of whenOptions kinda feel like they could be static constants somewhere?

Reply:
👍